osm_fudge/index/bk_tree.py

```diff
@@ -0,0 +1,90 @@
+'''An implementation of bk-tree
+
+More: https://en.wikipedia.org/wiki/BK-tree
+'''
+
+
+def grouped(iterable, n):
+    return zip(*[iter(iterable)]*n)
+
+
+class BKTree:
+    '''A bk-tree implementation and defined user methods
+       for insertion and querying the string
+    '''
+
+    def __init__(self, distanceMetrics):
+        self.tree = [[]]
+        self.distanceMetrics = distanceMetrics
+
+    def insert_bk_element(self, level_index, str, dist):
+        self.tree[level_index].append(dist)
+        self.tree[level_index + 1].append(str)
+        sibling_index = self.tree[level_index + 1].index(str)
```

Comment:
Hm `str` is a global Python function (for converting to string), not sure if it's a good idea to name your vars like that.